src/form/acr.jl

```diff
@@ -269,8 +368,78 @@ function constraint_mc_load_power_balance(pm::_PM.AbstractACRModel, nw::Int, i::
         push!(cstr_q, cq)
     end
 
-    con(pm, nw, :lam_kcl_r)[i] = cstr_p
-    con(pm, nw, :lam_kcl_i)[i] = cstr_q
+    con(pm, nw, :lam_kcl_r)[i] = isa(cstr_p, Array) ? cstr_p : [cstr_p]
+    con(pm, nw, :lam_kcl_i)[i] = isa(cstr_q, Array) ? cstr_q : [cstr_q]
+
+    if _IM.report_duals(pm)
+        sol(pm, nw, :bus, i)[:lam_kcl_r] = cstr_p
+        sol(pm, nw, :bus, i)[:lam_kcl_i] = cstr_q
+    end
+end
+
+
+""
+function constraint_mc_shed_power_balance(pm::_PM.AbstractACRModel, nw::Int, i::Int, bus_arcs, bus_arcs_sw, bus_arcs_trans, bus_gens, bus_storage, bus_pd, bus_qd, bus_gs, bus_bs)
```

Comment:
In PMR this is called `constraint_power_balance_shed`, consider moving the location of `shed`.

Reply:
I will make a new Issue for this